res/controllers/Numark-Party-Mix.scripts.js

```diff
@@ -0,0 +1,409 @@
+    midi.sendShortMsg(0x84, 0x16, ledOff);
+    midi.sendShortMsg(0x84, 0x17, ledOff);
+    midi.sendShortMsg(0x85, 0x14, ledOff);
+    midi.sendShortMsg(0x85, 0x15, ledOff);
+    midi.sendShortMsg(0x85, 0x16, ledOff);
+    midi.sendShortMsg(0x85, 0x17, ledOff);
+
+    // dim LEDs for scratch buttons
+    midi.sendShortMsg(0x90, 0x07, ledOff);
+    midi.sendShortMsg(0x91, 0x07, ledOff);
+
+    // untoggle (dim) PFL switches
+    midi.sendShortMsg(0x80, 0x1B, ledOff);
+    midi.sendShortMsg(0x81, 0x1B, ledOff);
+};
+
+NumarkPartyMix.Deck = function(deckNumber) {
+    components.Deck.call(this, deckNumber);
+
+    var channel = deckNumber - 1;
+    var deck = this;
+    this.scratchModeEnabled = false;
+
+    this.playButton = new components.PlayButton({
+        midi: [0x90 + channel, 0x00],
+    });
+
+    this.cueButton = new components.CueButton({
+        midi: [0x90 + channel, 0x01],
+    });
+
+    this.syncButton = new components.SyncButton({
+        midi: [0x90 + channel, 0x02],
+    });
+
+    this.headphoneButton = new components.Button({
+        midi: [0x90 + channel, 0x1B],
+        group: "[Channel" + deckNumber + "]",
+        inKey: "pfl",
+        outKey: "pfl",
+        output: function() {
+            if (engine.getParameter("[Channel" + deckNumber + "]", "pfl") === 1) {
+                midi.sendShortMsg(0x90 + channel, 0x1B, 0x7F);
+            } else {
+                midi.sendShortMsg(0x80 + channel, 0x1B, 0x00);
+            }
+        }
+    });
+
+    this.loadButton = new components.Button({
+        inKey: "LoadSelectedTrack"
+    });
+
+    this.volume = new components.Pot({
+        inKey: "volume"
+    });
+
+    this.gain = new components.Pot({
+        inKey: "pregain"
+    });
+
+    this.treble = new components.Pot({
+        group: "[EqualizerRack1_" + this.currentDeck + "_Effect1]",
+        inKey: "parameter3"
+    });
+
+    this.bass = new components.Pot({
+        group: "[EqualizerRack1_" + this.currentDeck + "_Effect1]",
+        inKey: "parameter1"
+    });
+
+    this.pitch = new components.Pot({
+        inKey: "rate",
+        invert: false
+    });
+
+    this.padSection = new NumarkPartyMix.PadSection(deckNumber);
+
+    this.scratchToggle = new components.Button({
+        midi: [0x90 + channel, 0x07],
+        input: function(channel, control, value, _status, group) {
+            if (!this.isPress(channel, control, value)) {
+                return;
+            }
+            deck.scratchModeEnabled = !deck.scratchModeEnabled;
+            if (deck.scratchModeEnabled) {
+                engine.scratchEnable(script.deckFromGroup(group), NumarkPartyMix.jogScratchSensitivity, 33 + 1 / 3, NumarkPartyMix.jogScratchAlpha, NumarkPartyMix.jogScratchBeta, true);
+            } else {
+                engine.scratchDisable(script.deckFromGroup(group));
+            }
+            // change the scratch mode status light
+            this.send(deck.scratchModeEnabled ? this.on : this.off);
```

Comment:
I'm not sure this works as intended. What physical control is this mapped to? If you scratch enable without actually moving the wheel, mixxx will think you just touched the platter, effectively stopping playback. 
Does the controller have a touch-sensitive platter?

Reply:
No it isn't touch sensitive. I don't scratch so I don't know what would be expected.

Reply:
Good question. In this case, I'd enable the scratching only once the platter is moved and disable the scratching once it has been moved the last time in a couple milliseconds. 
Another possible implementation would be to `jog` by default and only scratch when the scratch button is held down, but that likely feels a bit awkward too.
How does platter work work when the controller is used with other software?

Reply:
your first suggestion makes most sense but not sure when i will implement it.
I don't know how the controller works with other software.

Reply:
From the User Guide:
```quote
Platter/Jog Wheel: This jog wheel controls the audio when the wheel is moved. When
the Scratch button is not active, use the jog wheel to bend the pitch of the track. When
the Scratch button is active, use the jog wheel to grab and move the audio, “scratching”
the track as you would with a vinyl record.
```
So I guess the former suggestion would be closer to the "reference".

Reply:
I will not implement it soon.